modules/system/console/MixCompile.php

```diff
@@ -124,20 +135,26 @@ protected function canCompilePackage(string $mixJsPath): bool
     {
         if (!isset($this->packageJson)) {
             // Load the main package.json for the project
-            $canModifyPackageJson = null;
-            $packageJsonPath = base_path('package.json');
-            $packageJson = [];
-            if (File::exists($packageJsonPath)) {
-                $packageJson = json_decode(File::get($packageJsonPath), true);
-            }
-            $this->packageJson = $packageJson;
+            $this->packageJson = $this->readNpmPackageManifest();
         }
 
         $workspacesPackages = $this->packageJson['workspaces']['packages'] ?? [];
 
         return in_array($this->getPackagePath($mixJsPath), $workspacesPackages);
     }
 
+    /**
+     * Read the package.json file for the project, path configurable with the
+     * `NPM_PACKAGE_MANIFEST` env option
+     */
+    protected function readNpmPackageManifest(): array
+    {
+        $packageJsonPath = base_path(env('NPM_PACKAGE_MANIFEST', 'package.json'));
```

Comment:
Thinking on it more @jaxwilko this would be better as an option rather than an env() variable. Probably `--manifest` would make the most sense.

Reply:
Added in https://github.com/wintercms/winter/pull/642/commits/2b0228b45afaa22e7a30427fc0a16464f7365d91